src/main/java/org/broadinstitute/consent/http/db/ElectionDAO.java

```diff
@@ -106,82 +131,114 @@ void updateElectionStatus(@BindList("electionsId") List<Integer> electionsId,
-            +" AND lower(e.electionType) = lower(:type)  order by createDate asc")
+    @SqlQuery(
+        "SELECT DISTINCT " +
+            "e.election_id, e.dataset_id, v.vote final_vote, e.status, e.create_date, e.reference_id, v.rationale final_rationale, v.createDate final_vote_date, " +
+            "e.last_update, e.final_access_vote, e.election_type, e.data_use_letter, e.dul_name, e.archived, e.version " +
+        "FROM election e " +
+        "INNER JOIN vote v ON v.electionId = e.election_id AND LOWER(v.type) = 'final' " +
+        "INNER JOIN " +
+            "(SELECT reference_id, MAX(create_date) max_date " +
+            "FROM election e " +
+            "WHERE e.election_type = :type " +
+            "GROUP BY reference_id) election_view " +
+                "ON election_view.max_date = e.create_date " +
+                "AND election_view.reference_id = e.reference_id " +
+                "AND LOWER(e.election_type) = LOWER(:type) " +
+        "ORDER BY create_date ASC")
     List<Election> findLastElectionsWithFinalVoteByType(@Bind("type") String type);
 
-    @SqlQuery(" SELECT DISTINCT e.electionid, e.datasetid, v.vote finalvote, e.status, e.createdate, e.referenceid, " +
-            "   v.rationale finalrationale, v.createdate finalvotedate, " +
-            "   e.lastupdate, e.finalaccessvote, e.electiontype, e.datauseletter, e.dulname, e.archived, e.version " +
+    @SqlQuery(
+        "SELECT DISTINCT e.election_id, e.dataset_id, v.vote final_vote, e.status, e.create_date, e.reference_id, " +
+            "   v.rationale final_rationale, v.createdate final_vote_date, " +
+            "   e.last_update, e.final_access_vote, e.election_type, e.data_use_letter, e.dul_name, e.archived, e.version " +
             " FROM election e " +
-            " INNER JOIN vote v ON v.electionid = e.electionid " +
+            " INNER JOIN vote v ON v.electionid = e.election_id " +
             " INNER JOIN " +
-            "   (SELECT e2.referenceid, MAX(e2.createDate) maxdate " +
+            "   (SELECT e2.reference_id, MAX(e2.create_date) max_date " +
             "    FROM election e2 " +
-            "    WHERE LOWER(e2.electiontype) = 'dataaccess' " +
+            "    WHERE LOWER(e2.election_type) = 'dataaccess' " +
             "    AND LOWER(e2.status) = LOWER(:status) " +
-            "    GROUP BY e2.referenceid) electionview " +
-            "       ON electionview.maxdate = e.createdate " +
-            "       AND electionview.referenceid = e.referenceid" +
-            " WHERE LOWER(e.electiontype) = 'dataaccess' " +
-            " AND e.finalAccessVote IS true " +
+            "    GROUP BY e2.reference_id) election_view " +
+            "       ON election_view.max_date = e.create_date " +
+            "       AND election_view.reference_id = e.reference_id" +
+            " WHERE LOWER(e.election_type) = 'dataaccess' " +
+            " AND e.final_access_vote IS true " +
```

Comment:
Observation: This condition is something that we should have in other final access queries as well, but that's very much outside the scope of this PR. When we create votes for an election, there is a chairperson vote type as well as a final vote type, so two votes per chairperson. I can see cases where we might get a cartesian product of votes in either case.